### PR TITLE
Adds the function app setup

### PR DESCRIPTION
**Pull request by orta**
Will highlight the important stuff inline now

### CODE REVIEW

File: .github/workflows/upload.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - name: 'run npm'
+      run: |
+        npm install
+        npm run build --if-present
+        npm run test --if-present
+    - uses: Azure/functions-action@v1
+      id: fa
+      with:
+        app-name: TypeScriptReposAutomation
+        publish-profile: ${{ secrets.SCM_CREDENTIALS }}
```
Comment (orta) [inline .github/workflows/upload.yml]: This _should_ automatically deploy the app on every push to master, I've set up all the credentials etc

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "configurations": [
+    {
+      "name": "Attach to Node Functions",
+      "type": "node",
+      "request": "attach",
+      "port": 9229,
+      "preLaunchTask": "func: host start"
+    }
+  ]
+}
```
Comment (orta) [inline .vscode/launch.json]: I've validated that this correctly loads up a debugger (though it did do it on the 2nd API call, not the first)

File: TypeScriptRepoIssueWebhook/index.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        // };
+        // return;
+    // }
+
+
+    context.res = {
+        body: "Hello there are " + numberOfThings + "things"
+    };
+};
+
```
Comment (orta) [inline TypeScriptRepoIssueWebhook/index.ts]: This function is what runs - I validated a npm module and relative link

File: fixtures/issues/created.json [hunk 4]
```diff
@@ -0,0 +1,185 @@
+{
```
Comment (orta) [inline fixtures/issues/created.json]: Coming up are a bunch of fixture webhooks

File: shared/constants.test.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { numberOfThings } from "./constants"
+
+it("has the rught number", () => {
+  expect(numberOfThings).toEqual(23)
+})
```
Comment (orta) [inline shared/constants.test.ts]: It's a test, so Jest won't fail when you first do it

**Comment by glaucia86** [comment]
Hi @orta

Another good thing that you can integrate this Function App to help even more this automatization is integrate with the Azure tracking using the Azure Boards [Here](https://docs.microsoft.com/en-us/azure/devops/boards/index?view=azure-devops)

**Comment by glaucia86** [comment]
There's an example 

* [HERE](https://github.com/microsoft/azure-devops-auth-samples/tree/master/JavascriptWebAppSample)

**Comment by orta** [comment]
Alright, I'm gonna merge in some of these

**Pull request merged**